### PR TITLE
Fix python build

### PR DESCRIPTION
**Pull request by ackRow**
- Fix conflicts between flutter and python build on MacOS
- `decrypt` function in Python now returns both plaintext and metadata like the `ffi`

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -121,7 +121,7 @@ build_windows:
     - cbindgen . -c cbindgen.toml | grep -v \#include | uniq >target/${CI_PROJECT_NAME}.h
     # python build
     - pip install -r src/interfaces/pyo3/tests/requirements.txt
-    - maturin build --release --features python --target x86_64-pc-windows-gnu
+    - maturin build --release --features python --target x86_64-pc-windows-gnu --target-dir target_python
   artifacts:
     paths:
       - target/x86_64-pc-windows-gnu/release/*.dll
```
Comment (Manuthor) [inline .gitlab-ci.yml:127]: we should keep as artifact the `target_python` folder

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -138,9 +138,9 @@ build_osx:
...
     - cargo build --release --features ffi --target x86_64-apple-darwin
     - cargo lipo --release --features ffi
-    # python build
+    # python build: custom `target_dir` to avoid conflicts with flutter
     - pip3 install -r src/interfaces/pyo3/tests/requirements.txt
-    - maturin build --release --features python --target x86_64-apple-darwin
+    - maturin build --release --features python --target x86_64-apple-darwin --target-dir target_python
   artifacts:
     paths:
       - target/x86_64-apple-darwin/release/*.dylib
```
Comment (Manuthor) [inline .gitlab-ci.yml:146]: let's keep whatever we need from `target_python`

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -166,7 +166,7 @@ build_python_manylinux:
     entrypoint: ['']
   script:
     # Build for glibc 2.17
-    - maturin build --compatibility manylinux_2_17 --release --features python
+    - maturin build --compatibility manylinux_2_17 --release --features python --target-dir target_python
   artifacts:
     paths:
       - target/wheels/*manylinux*.whl
```
Comment (Manuthor) [inline .gitlab-ci.yml:172]:
```suggestion
      - target_python/wheels/*manylinux*.whl
```

File: src/interfaces/pyo3/py_cover_crypt.rs [hunk 4]
```diff
@@ -269,14 +266,14 @@ impl CoverCrypt {
     /// - `encrypted_header_bytes`  : encrypted header bytes
     /// - `authentication_data`      : authentication data to use in symmetric decryption
     ///
-    /// Returns: (SymmetricKey, plaintext bytes)
+    /// Returns: (SymmetricKey, metadata bytes)
```
Comment (Manuthor) [inline src/interfaces/pyo3/py_cover_crypt.rs:269]:
```suggestion
    /// Returns: (SymmetricKey, header metadata bytes)
```

File: src/interfaces/pyo3/tests/test_cover_crypt.py [hunk 5]
```diff
@@ -169,10 +169,11 @@ def test_simple_encryption_decryption(self) -> None:
         )
 
         # Successful decryption
-        cleartext = self.cc.decrypt(
+        cleartext, metadata = self.cc.decrypt(
```
Comment (Manuthor) [inline src/interfaces/pyo3/tests/test_cover_crypt.py:172]:
```suggestion
        cleartext, header_metadata = self.cc.decrypt(
```

File: src/interfaces/pyo3/tests/test_cover_crypt.py [hunk 6]
```diff
@@ -231,10 +232,10 @@ def test_policy_rotation_encryption_decryption(self) -> None:
             keep_old_accesses=True,
         )
 
-        cleartext = self.cc.decrypt(
+        cleartext, _ = self.cc.decrypt(
```
Comment (Manuthor) [inline src/interfaces/pyo3/tests/test_cover_crypt.py:235]:
```suggestion
        plaintext, _ = self.cc.decrypt(
```

File: src/interfaces/pyo3/tests/test_cover_crypt.py [hunk 7]
```diff
@@ -249,7 +250,7 @@ def test_policy_rotation_encryption_decryption(self) -> None:
                 sec_high_fr_sp_user, ciphertext, self.authenticated_data
             )
 
-        cleartext = self.cc.decrypt(
+        cleartext, _ = self.cc.decrypt(
```
Comment (Manuthor) [inline src/interfaces/pyo3/tests/test_cover_crypt.py:253]:
```suggestion
        plaintext, _ = self.cc.decrypt(
```

File: .gitlab-ci.yml (hunk 1 above)
Comment (ackRow) [reply .gitlab-ci.yml:127]: actually we don't need the libs from `./target_python`, the resulting .whl package is saved in `./target`

**Pull request merged**